ui/src/app/base/components/node/StorageTables/DatastoresTable/DatastoresTable.test.tsx

```diff
@@ -16,105 +20,150 @@ import {
 
 const mockStore = configureStore();
 
-describe("DatastoresTable", () => {
-  it("shows a message if no datastores are detected", () => {
-    const notDatastore = fsFactory({ fstype: "fat32" });
-    const state = rootStateFactory({
-      machine: machineStateFactory({
-        items: [
-          machineDetailsFactory({
-            disks: [
-              diskFactory({ name: "not-datastore", filesystem: notDatastore }),
-            ],
-            system_id: "abc123",
-          }),
-        ],
-      }),
-    });
-    const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <DatastoresTable canEditStorage systemId="abc123" />
-      </Provider>
-    );
-    expect(wrapper.find("[data-testid='no-datastores']").text()).toBe(
-      "No datastores detected."
-    );
+it("shows a message if no datastores are detected", () => {
+  const notDatastore = fsFactory({ fstype: "fat32" });
+  const machine = machineDetailsFactory({
+    disks: [diskFactory({ name: "not-datastore", filesystem: notDatastore })],
+    system_id: "abc123",
+  });
+  const state = rootStateFactory({
+    machine: machineStateFactory({
+      items: [machine],
+    }),
   });
+  const store = mockStore(state);
+  render(
```

Comment:
Just a note: I've been using the `renderWithMockStore` and `renderWithBrowserRouter` utils when refactoring the tests, partially so that it's easier to remove `CompatRouter` when we've migrated to React Router v6.

https://github.com/canonical-web-and-design/maas-ui/blob/main/ui/src/testing/utils.tsx#L135-L170